src/extension.ts

```diff
@@ -25,8 +25,8 @@ import * as httpgdViewer from './plotViewer';
 import { RMarkdownPreviewManager } from './rmarkdown/preview';
 
 // global objects used in other files
-export const extDir: string = path.join(os.homedir(), '.vscode-R');
-export const tmpDir: string = path.join(extDir, 'tmp');
+export const extensionDirectory: string = path.join(os.homedir(), '.vscode-R');
+export const temporaryDirectory: string = path.join(extensionDirectory, 'tmp');
```

Comment:
It looks like `tmp` or `temp` are already a convention to indicate "temporary` (e.g. `os.tmpdir()`). Not sure if it really looks better to call it in full.

Reply:
This is a bit ambiguous too, i.e. `temporaryDirectory` just looks like an alias of `os.tmpdir()`?

Reply:
Would it be feasible to simply use `os.tmpdir()` then (if I recall correctly, there might be issues with user privacy on some platforms)?

And if not, why is this directory declared/created here? It doesn't seem to be used anywhere eles in the code, so wouldn't it be enough to put this in the markdown-preview code?

Reply:
Using `os.tmpdir()` will have some user privacy problem, unless we create something like `/tmp/vscode-R/$USER` and make the folder have mode 700.

Just notice that https://github.com/REditorSupport/vscode-R/blob/master/src/helpViewer/helpProvider.ts#L289 uses `os.tmpdir()` which is probably subject to multi-user clash problem.

Reply:
Perhaps we should set all references of os.tmpdir (and future references) to temporaryDirectory*?

* barring issues with the name